test/functionalTest/cases/2998_null_not_working_in_q_and_mq_filter/null_not_working_in_mq_for_subscription.test

```diff
@@ -0,0 +1,258 @@
+# Copyright 2017 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+# VALGRIND_READY - to mark the test ready for valgrindTestSuite.sh
+
+--NAME--
+MQ filter in subscription with null value of attribute
+
+--SHELL-INIT--
+dbInit CB
+brokerStart CB
+accumulatorStart --pretty-print
+
+--SHELL--
+
+#
+# 01. Create subscription for entity E*, attribute 't' with a filter mq=t.m==null
+# 02. Create entity E1 with attr 't' having a metadata 'm' == null
+# 03. Make sure ONE notification has been sent
+# 04. Modify E1::t.m to 4
+# 05. Make sure no notification has been sent
+# 06. Modify E1::t.m to null
+# 07. Make sure ONE notification has been sent
```

Comment:
The following for 'null as string' could be added to make the test stronger:

```
# 07. Modify E1::t.m to 'null'
# 08. Make sure no notification has been sent
```

Reply:
Fixed in ba413e82e